### ISSUE TITLE
How to nest contents in separate folders? (Previous manual solution doesn't work anymore) 

### ISSUE THREAD

**Issue by ErwinTATP** [panr/hugo-theme-terminal#191]
I'm afraid I have to bother you with an age-old question: how to keep contents in separate folders. 

I tried to organize my website contents like this:

  ```
/content
|--post/
        |--post1.md
        |--post2.md
|--notes/ 
        |-post3.md
        |-post4.md
|--books/
        |-post5.md
        |-post6.md
```

The problem is only the post1 and 2 in "post" folder (that I've specified for `contentTypeName`) will be pushed to the website, while the posts 3\~6 seemed to be ignored by hugo.

Readme.md reminded me that nesting in this theme is currently difficult, but I found Issues #23 and #97 seems to provide a possible manual fix. Yet the codelines mentioned in the reply (change `$.Site.Params.contentTypeName` etc.)has definitely changed a lot over the years.

I looked into the repositories and found the said codelines in list.html [had been migrated to index.html](https://github.com/panr/hugo-theme-terminal/commit/0925abe5d8b91911c007b292e1a4831299c8f3c2). Now I have no idea how to make nesting work with the new templates.

**Comment by panr** [comment]
So, for example, `localhost:1313/notes/post3` doesn't work? What you mean by:

>  posts 3~6 seemed to be ignored by hugo

?

**Comment by ErwinTATP** [comment]
Exactly, if I try `localhost:1313/notes/post3` `localhost:1313/notes/post4` `localhost:1313/books/post5` `localhost:1313/books/post6` I can only get 404 pages.

**Comment by panr** [comment]
No errors nowehere? I see that your `notes/**` don't have `.Title` param, maybe that's why?

**Comment by ErwinTATP** [comment]
Very likely. Where can I set that up? (really sorry I'm new to hugo)

**Comment by ErwinTATP** [comment]
nvm I figured it out.

- Under `layouts` I created folders `notes`, `books`, etc
- In each of these new folders, I copied the `index.html` and `list.html` from `layouts/_default`.
- Then I changed the parameters in the newly copied `index.html` according to #97 : for example, in `layouts/notes/index.html` I replaced every `$.Site.Params.contentTypeName` into `$.Site.Params.noteSection` (`noteSection` is my placeholder for `/notes` posts). 
- Still in the new `index.html`, find the line
```
{{ $paginator := .Paginate (where $PageContext.RegularPages "Type" $contentTypeName) }}
```
and change the `$contentTypeName` accordingly (eg. `{{ $paginator := .Paginate (where $PageContext.RegularPages "Type" "notes") }}` ). Now my `layouts/notes/index.html` looks like this: (line 8\~15)
```
    {{ $isntDefault := not (or (eq (trim $.Site.Params.noteSection " ") "posts") (eq (trim $.Site.Params.noteSection " ") "")) }}
    {{ $contentTypeName := cond $isntDefault (string $.Site.Params.noteSection) "posts" }}

    {{ $PageContext := . }}
    {{ if .IsHome }}
      {{ $PageContext = .Site }}
    {{ end }}
    {{ $paginator := .Paginate (where $PageContext.RegularPages "Type" "notes") }}
```

- Finally, in config.toml, add `noteSection = "notes"`

Thats it; the contents under `localhost:1313/notes` and `localhost:1313/books` now works perfectly.

I don't know if every step I listed is necessary, but it works. 

Thank you for your patient answer and amazing hugo template!
